WMF Framework/Remote Notifications/Model/RemoteNotificationsModelController.swift

```diff
@@ -0,0 +1,221 @@
+@objc enum RemoteNotificationsModelChangeType: Int {
+    case addedNewNotifications
+}
+
+@objc final class RemoteNotificationsModelChange: NSObject {
+    @objc let type: RemoteNotificationsModelChangeType
+    @objc let notifications: [RemoteNotification]
+
+    init(type: RemoteNotificationsModelChangeType, notifications: [RemoteNotification]) {
+        self.type = type
+        self.notifications = notifications
+        super.init()
+    }
+}
+
+@objc final class RemoteNotificationsModelChangeResponseCoordinator: NSObject {
+    @objc let modelChange: RemoteNotificationsModelChange
+    private let modelController: RemoteNotificationsModelController
+
+    init(modelChange: RemoteNotificationsModelChange, modelController: RemoteNotificationsModelController) {
+        self.modelChange = modelChange
+        self.modelController = modelController
+        super.init()
+    }
+
+    @objc func markAsRead(_ notification: RemoteNotification) {
+        modelController.markAsRead(notification)
+    }
+}
+
+@objc final class RemoteNotificationsModelController: NSObject {
+    @objc static let ModelDidChangeNotification = NSNotification.Name(rawValue: "RemoteNotificationsModelDidChangeNotification")
+
+    let managedObjectContext: NSManagedObjectContext
+
+    required override init() {
+        let modelName = "RemoteNotifications"
+        let modelExtension = "momd"
+        let modelBundle = Bundle.wmf
+        guard let modelURL = modelBundle.url(forResource: modelName, withExtension: modelExtension) else {
+            assertionFailure("Couldn't find url for resource named \(modelName) with extension \(modelExtension) in bundle \(modelBundle); make sure you're providing the right name, extension and bundle")
+            abort() // TODO
+        }
+        guard let model = NSManagedObjectModel(contentsOf: modelURL) else {
+            assertionFailure("Couldn't create model with contents of \(modelURL); make sure \(modelURL) is the correct url for \(modelName)")
+            abort() // TODO
+        }
+        let container = NSPersistentContainer(name: modelName, managedObjectModel: model)
+        let sharedAppContainerURL = FileManager.default.wmf_containerURL()
+        let remoteNotificationsStorageURL = sharedAppContainerURL.appendingPathComponent(modelName)
+        let description = NSPersistentStoreDescription(url: remoteNotificationsStorageURL)
+        container.persistentStoreDescriptions = [description]
+        container.loadPersistentStores { (storeDescription, error) in
+            if let error = error {
+                fatalError("Unexpected Core Data error occurred while loading persistent stores: \(error)") // TODO
```

Comment:
Not sure what to do in this case, I could post a notification and stop the controller if this fails

Reply:
Ensuring the controller is stopped and stays stopped seems like a good plan. I would also suggest removing the fatalError so the app doesn't crash